diff.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+import json
+
+
+def compare_objects(obj1, obj2):
+    diff_funcs = []
+    if obj1['name'] == obj2['name']:
+        if obj1['functions'] != obj2['functions'] or obj1['coverage'] != obj2['coverage']:
+            if obj1['functions'] == obj2['functions']:
+                obj1['functions'] = []
+            if obj1['functions'] != obj2['functions']:
+                for func1 in obj1['functions']:
+                    for func2 in obj2['functions']:
+                        if func1['name'] == func2['name']:
+                            if func1['exec'] is False and func2['exec'] is True:
+                                func1['exec'] = True
+                                diff_funcs.append(func1)
+                obj1['functions'] = diff_funcs
+            list_val = []
+            for i, j in zip(obj1['coverage'], obj2['coverage']):
+                if j is not None and j is not 0 and j >= i:
+                    i = j - i
+                list_val.append(i)
+            obj1['coverage'] = list_val
+            return json.dumps(obj1)
+    return None
+
+
+def compare_reports(baseline_report, report):
+    baseline_coverage = []
+    coverage = []
+    for k, j in zip(baseline_report, report):
+        if k == 'source_files':
+            baseline_coverage = baseline_report[k]
+        if j == 'source_files':
+            coverage = report[j]
+    result = []
+    for i in range(0, len(baseline_coverage)):
```

Comment:
Why are you using a range loop here? You can iterate `baseline_coverage` directly (same for the following line).